### PR TITLE
Include data for BCG stellar-halo mass relation

### PR DESCRIPTION
**Pull request by FHusko**
This data includes observational measurements of BCG stellar masses in fixed (2D) apertures of 10, 30, 50 and 100 kpc, so they can be used to compare with SOAP outputs. Various conversions are described in the files.

### CODE REVIEW

**Comment by EvgeniiChaikin** [comment]
Hi @FHusko, thanks for this nice update!

For each paper in your dataset, can you please move multiple scripts that use different apertures to convert data into single files, such that one paper/author == one script/file? Then there will be ~3 times less code, and there will be less repeated code.

**Comment by EvgeniiChaikin** [comment]
Then in each script, you can write something like

```
for aperture in [10, 50, 100]:
   do ...
   ... convert ...
   output_filename = "AuthorYear_{aperture}kpc.hdf5"
   output_path = f"{output_directory}/{output_filename}"
   processed.write(filename=output_path)
```

**Comment by FHusko** [comment]
Thanks @EvgeniiChaikin for the comments! I have now done this, it should be much more condensed.

File: data/GalaxyStellarMassHaloMassBCG/conversion/convertDeMaio2020.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+h = h_sim
+
+# Write a separate file for each aperture
+apertures = [10, 50, 100]
+for x in range(3):
+    output_filename = f"DeMaio2020_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
Comment (EvgeniiChaikin) [inline data/GalaxyStellarMassHaloMassBCG/conversion/convertDeMaio2020.py:42]:
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [10, 50, 100]  # kpc

# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"DeMaio2020_{aperture}kpc.hdf5"
```

File: data/GalaxyStellarMassHaloMassBCG/conversion/convertGoldenMarx2023.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+# Write a separate file for each aperture
+apertures = [10, 30, 50, 100]
+for x in range(4):
+
+    output_filename = f"GoldenMarx2023_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
Comment (EvgeniiChaikin) [inline data/GalaxyStellarMassHaloMassBCG/conversion/convertGoldenMarx2023.py:41]:
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [10, 30, 50, 100]  # kpc
    
# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"GoldenMarx2023_{aperture}kpc.hdf5"
```

File: data/GalaxyStellarMassHaloMassBCG/conversion/convertHuang2022.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+# Write a separate file for each aperture
+apertures = [10, 30, 50, 100]
+for x in range(4):
+
+    output_filename = f"Huang2022_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
Comment (EvgeniiChaikin) [inline data/GalaxyStellarMassHaloMassBCG/conversion/convertHuang2022.py:42]:
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [10, 30, 50, 100]  # kpc

# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"Huang2022_{aperture}kpc.hdf5"
```

File: data/GalaxyStellarMassHaloMassBCG/conversion/convertKravstov2018.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+h = h_sim
+
+# Write a separate file for each aperture
+apertures = [30, 50, 100]
+for x in range(3):
+    output_filename = f"Kravstov2018_{apertures[x]}.hdf5"
+    output_directory = "../"
+
+    if not os.path.exists(output_directory):
+        os.mkdir(output_directory)
```
Comment (EvgeniiChaikin) [inline data/GalaxyStellarMassHaloMassBCG/conversion/convertKravstov2018.py:40]:
```suggestion
output_directory = "../"
if not os.path.exists(output_directory):
    os.mkdir(output_directory)

apertures = [30, 50, 100]  # kpc

# Write a separate file for each aperture
for x, aperture in enumerate(apertures):
    output_filename = f"Kravstov2018_{aperture}kpc.hdf5"
```

File: data/GalaxyStellarMassHaloMassBCG/conversion/convertLin2017.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    "is based on measurements of Kravstov et al. (2018) (2018AstL...44....8K)."
+    " See their Fig. 1, which shows roughly flat or slightly rising profiles."
+)
+citation = "Lin et al. (2017)"
+bibcode = "(2017ApJ...851..139L)"
+name = "BCG stellar mass-halo mass relation at z=0.7"
+plot_as = "points"
+redshift = 0.7
+redshift_lower = 0.0
+redshift_upper = 1
```
Comment (EvgeniiChaikin) [inline data/GalaxyStellarMassHaloMassBCG/conversion/convertLin2017.py:55]:
```suggestion
redshift_upper = 1.0
```

File: data/GalaxyStellarMassHaloMassBCG/conversion/convertKravstov2018.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    "Halo masses were converted from an M_500 to M_200 definition by assuming "
+    "needed. an NFW profile and concentration of c=5, which leads to a "
+    "conversion factor of 1.29. Stellar masses were measured in 30, 50 and "
+    "100 kpc apertures."
+)
+citation = "Kravstov et al. (2018)"
+bibcode = "2018AstL...44....8K"
+name = "BCG stellar mass-halo mass relation at z=0.15"
+plot_as = "points"
+redshift = 0.15
```
Comment (EvgeniiChaikin) [inline data/GalaxyStellarMassHaloMassBCG/conversion/convertKravstov2018.py:30]: Do you also want to provide `redshift_lower` and `redshift_upper`, similar to how you do it in the other scripts?

**Review by EvgeniiChaikin — COMMENTED** [review]
Hi @FHusko,

I added a couple of small comments in the code. 

A few more comments, 

- Do all these papers define M200 in the same way? And how is M200 exactly defined? Is it M200crit, mean, according to Bryan Norman (1998), or something else?
- It is not clear whether stellar masses are 2D or deprojected. E.g. Gonzalez et al (2013) provide both 2D and 3D masses.

**Comment by EvgeniiChaikin** [comment]
Lastly, I'd also move the `comment` variable inside the loop and make it an f-string such that if, say, a file contains the data measured in 50 kpc apertures, then the comment would say so (instead of current line `"Stellar masses were measured in 10, 50 and 100 kpc apertures`, which makes we think that the file contains data measured in multiple apertures.).

**Comment by FHusko** [comment]
Thanks Evgenii, I implemented your latest comments. As for halo masses: all that use 200 or 500 define them using critical densities. I now specify that. One paper uses the overdensity definition from Bryan & Norman (1998), I also specify that. 

We agreed with Joop to convert these all to M200_crit for reasons of having them all on the same plot, rather than looking at too many plots.

**Comment by FHusko** [comment]
I added some small comments for the Golden-Marx et al. observations. They use a mean overdensity rather than a critical overdensity definition. I do convert to account for this in the data already, but forgot that I do. I have added details about that in the descriptions.

**Review by EvgeniiChaikin — APPROVED** [review]
Thanks, @FHusko! The scripts look ready to me; I am going to merge them.

**Comment by EvgeniiChaikin** [comment]
Just for the purpose of keeping full records in this PR, this is how these data look


stellar_mass_halo_mass_M200_all_100

**Pull request merged**